### PR TITLE
AEA-1078 Use Ganache Docker image for testing

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

- refactor OEF search docker image
- add Ganache Docker image.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: tests/common/docker_image.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+        """Get the image tag."""
+        return "fetchai/oef-search:0.7"
+
+    def check_skip(self):
+        """Check if the test should be skipped."""
+        if sys.version_info < (3, 7):
+            pytest.skip("Python version < 3.7 not supported by the OEF.")
+            return
+
+        if os.name == "nt":
```
Comment (DavidMinarsch) [inline tests/common/docker_image.py:192]: Shouldn't we also skip macOS?

Comment (marcofavorito) [reply tests/common/docker_image.py:192]: I just copy-pasted previous code

File: tests/common/docker_image.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class DockerImage(ABC):
+    """A class to wrap interatction with a Docker image."""
+
+    def __init__(self, client: docker.DockerClient):
+        """Initialize."""
+        self._client = client
```
Comment (DavidMinarsch) [inline tests/common/docker_image.py:49]: Problem is, we initiate before we skip. I think that won't work as docker won't be present on the platforms where we want to skip.

File: tests/conftest.py [hunk 3]
```diff
@@ -653,32 +528,43 @@ def network_node(
...
-    success = _wait_for_oef(max_attempts=max_attempts, sleep_rate=timeout)
 
+def _launch_image(image: DockerImage, timeout: float = 2.0, max_attempts: int = 10):
+    """
+    Launch image.
+
+    :param image: an instancoe of Docker image.
+    :return: None
+    """
+    image.check_skip()
```
Comment (DavidMinarsch) [inline tests/conftest.py:551]: I think the instantiation needs to happen after skip! Skip needs to be class method independent of docker

File: tests/test_contracts/test_base.py [hunk 4]
```diff
@@ -138,10 +138,14 @@ def test_get_instance_no_address_ethereum(dummy_contract):
...
 
-def test_get_deploy_transaction_ethereum(dummy_contract):
+def test_get_deploy_transaction_ethereum(
+    dummy_contract, ganache_addr, ganache_port, ganache
+):
     """Tests the deploy transaction classmethod for ethereum."""
     ethereum_crypto = crypto_registry.make(ETHEREUM)
-    ledger_api = ledger_apis_registry.make(ETHEREUM, address=ETHEREUM_DEFAULT_ADDRESS,)
+    ledger_api = ledger_apis_registry.make(
+        ETHEREUM, address=f"{ganache_addr}:{ganache_port}"
```
Comment (DavidMinarsch) [inline tests/test_contracts/test_base.py:147]: Provide constant `ETHEREUM_DEFAULT_ADDRESS` in conftest of this form

File: tests/common/docker_image.py (hunk 2 above)
Comment (marcofavorito) [reply tests/common/docker_image.py:49]: Yes!! You're right

File: .github/workflows/workflow.yml [hunk 5]
```diff
@@ -19,6 +19,8 @@ jobs:
         with:
           python-version: 3.6
       - name: Install dependencies (ubuntu-latest)
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS:
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:23]: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

File: .github/workflows/workflow.yml [hunk 6]
```diff
@@ -211,6 +211,8 @@ jobs:
         brew install protobuf
     - if: matrix.os == 'windows-latest'
       name: Install dependencies (windows-latest)
+      env:
+        ACTIONS_ALLOW_UNSECURE_COMMANDS: true
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:215]: From a GH pipeline output (https://pipelines.actions.githubusercontent.com/7sDcrH6ORgsV46IrjyGkkm9svY0T5k5MHqINOHfMTGdQaFcLxW/_apis/pipelines/1/runs/5432/signedlogcontent/66?urlExpires=2020-11-17T09%3A27%3A18.0737144Z&urlSigningMethod=HMACV1&urlSignature=oDQkk4wRl0waEP0wnQHvMbXHEZEPqyvpmEdkxhKgn5U%3D):

```
2020-11-17T09:09:24.3667593Z ##[error]Unable to process command '::add-path::C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x64' successfully.
2020-11-17T09:09:24.3698299Z ##[error]The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
```

https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**